include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -90,12 +111,20 @@ namespace LinearAlgebra
                            const MPI_Comm  communicator,
                            const bool      omit_zeroing_entries)
     {
-      Teuchos::RCP<MapType> input_map =
+      // release memory before reallocation
+      vector.reset();
+      locally_owned_map.reset();
+      locally_relevant_map.reset();
+
+      // here we assume, that the index set is non-overlapping
+      compressed = true;
+      locally_owned_map =
         parallel_partitioner.make_tpetra_map_rcp(communicator, false);
 
-      if (vector->getMap()->isSameAs(*input_map) == false)
-        vector = Utilities::Trilinos::internal::make_rcp<VectorType>(input_map);
-      else if (omit_zeroing_entries == false)
+      if (!vector->getMap()->isSameAs(*locally_owned_map))
+        vector = Utilities::Trilinos::internal::make_rcp<VectorType>(
+          locally_owned_map);
+      if (!omit_zeroing_entries)
         {
           vector->putScalar(0.);
         }
```

Comment:
This is a `reinit()` function. Don't you want to reset the `source_stored_elements` object as well? Same in the other `reinit()` functions.

Reply:
Now that `reinit()` function also resets `source_stored_elements`